File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int maxp = 0, minp = prices[0];
        for(int i = 0; i<n; i++){
            maxp = max(maxp, prices[i]-minp);
            minp = min(minp, prices[i]);
        }
        return maxp;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        maxp = 0
        minp = prices[0]
        for i in range(n):
            maxp = max(maxp, prices[i]-minp)
            minp = min(minp, prices[i])
        return maxp